jit/src/main/java/org/dcsa/conformance/standards/jit/JitScenarioListBuilder.java

```diff
@@ -1,81 +1,280 @@
+                                .then(
+                                    sendTimestamp(context, JitTimestampType.ESTIMATED)
+                                        .then(
+                                            portCallService(context, serviceType)
+                                                .then(
+                                                    sendTimestamp(
+                                                            context, JitTimestampType.REQUESTED)
+                                                        .then(
+                                                            sendTimestamp(
+                                                                    context,
+                                                                    JitTimestampType.PLANNED)
+                                                                .then(
+                                                                    sendTimestamp(
+                                                                        context,
+                                                                        JitTimestampType
+                                                                            .ACTUAL)))))))
+                    .toList()
+                    .toArray(new JitScenarioListBuilder[] {})));
+
+    // Scenario: "S(service type) - E - R - S - P - A"
+    // Scenario: "S(service type) - E - R - P - S - A"
+    // Scenario: "S(service type) - E - R - P - A - S"
+
+    // Scenario group: "S-ERP-A out-of-band variations"
+
+    // Scenario group: "S-A variations"
+
+    // Scenario group: "S-ERP-A cancel"
+
+    // Scenario group: "S-ERP-A decline"
+
+    // Scenario suite: "Secondary sender and receiver conformance"
+
+    // Scenario group: "S-ERP-A as FYI messages"
+    //     Scenario group: "S-ERP-A as FYI messages"
+    //            Scenario: "S(service type) - E - R(forwarded by provider) - P - A"
+    //            Scenario: "S(service type) - E - R(forwarded by provider) - P - C"
+    //            Scenario: "S(service type) - E - R(forwarded by provider) - P - D(forwarded by
+    // provider)"
+    //
+    //    Scenario group: "S-A as FYI messages"
+    //            Scenario: "S(service type) - A"
+    //    Scenario group: "S as FYI message"
+    //        Scenario: "S(Moves)"
+    return scenarioList;
   }
 
-  private JitScenarioListBuilder(Function<ConformanceAction, ConformanceAction> actionBuilder) {
-    super(actionBuilder);
+  private static JitScenarioListBuilder sendTimestamp(
+      JitScenarioContext context, JitTimestampType timestampType) {
+    if (timestampType == JitTimestampType.REQUESTED) {
+      return new JitScenarioListBuilder(
+          previousAction ->
+              new JitTimestampAction(context, previousAction, JitTimestampType.REQUESTED, false));
+    }
+    return new JitScenarioListBuilder(
+        previousAction -> new JitTimestampAction(context, previousAction, timestampType, true));
```

Comment:
Looks like the only delta between these two branches is the last parameter. If so, consider simplifying to computing the boolean and just passing it rather than doing almost identical branches for a boolean parameter (less mental load to understand what is going on).

Reply:
Yes, you're right. Changing this, thanks.